connectors/sources/redis.py

```diff
@@ -0,0 +1,432 @@
+        self._logger = logger
+        self.host = self.configuration["host"]
+        self.port = self.configuration["port"]
+        self.database = self.configuration["database"]
+        self.username = self.configuration["username"]
+        self.password = self.configuration["password"]
+        self.ssl_enabled = self.configuration["ssl_enabled"]
+        self.mutual_tls_enabled = self.configuration["mutual_tls_enabled"]
+        self.tls_certfile = self.configuration["tls_certfile"]
+        self.tls_keyfile = self.configuration["tls_keyfile"]
+        self._redis_client = None
+        self.cert_file = ""
+        self.key_file = ""
+
+    def set_logger(self, logger_):
+        self._logger = logger_
+
+    def store_ssl_key(self, key, suffix):
+        if suffix == ".key":
+            pem_certificates = get_pem_format(
+                key=key, postfix="-----END RSA PRIVATE KEY-----"
+            )
+        else:
+            pem_certificates = get_pem_format(key=key)
+        with NamedTemporaryFile(mode="w", suffix=suffix, delete=False) as cert:
+            cert.write(pem_certificates)
+            return cert.name
+
+    def remove_temp_files(self):
+        for file_path in [self.cert_file, self.key_file]:
+            if os.path.exists(file_path):
+                try:
+                    os.remove(file_path)
+                except Exception as exception:
+                    self._logger.warning(
+                        f"Something went wrong while removing temporary certificate file: {file_path}. Exception: {exception}",
+                        exc_info=True,
+                    )
+
+    @cached_property
+    def _client(self):
+        if self.ssl_enabled and self.mutual_tls_enabled:
+            self.cert_file = self.store_ssl_key(key=self.tls_certfile, suffix=".crt")
+            self.key_file = self.store_ssl_key(key=self.tls_keyfile, suffix=".key")
+            query = f"ssl_certfile={self.cert_file}&ssl_keyfile={self.key_file}"
+            self._redis_client = redis.from_url(
+                f"rediss://{self.username}:{self.password}@{self.host}:{self.port}?{query}",
+                decode_responses=True,
+            )
+        elif self.ssl_enabled:
+            self._redis_client = redis.from_url(
+                f"rediss://{self.username}:{self.password}@{self.host}:{self.port}",
+                decode_responses=True,
+            )
+
+        else:
+            self._redis_client = redis.from_url(
+                f"redis://{self.username}:{self.password}@{self.host}:{self.port}",
+                decode_responses=True,
+            )
+        return self._redis_client
+
+    async def close(self):
+        if self._redis_client:
+            await self._client.aclose()  # pyright: ignore
+        self.remove_temp_files()
+
+    async def validate_database(self, db):
+        try:
+            await self._client.execute_command("SELECT", db)
+            await self._client.ping()
```

Comment:
Should this be the other way around? You first ping and then execute a dummy select?

Reply:
Actually this line is not making much sense here: ` await self._client.ping()` because if database is not valid then line `109` will throw exception else not.. so this line can be removed.